project_forecast_line/tests/test_forecast_line.py

```diff
@@ -489,39 +490,91 @@ def setUpClass(cls):
-                forecast.employee_resource_forecast_line_id.consolidated_forecast,
-                0.25,
-            )
+        self.env["hr.employee.forecast.role"].create(
+            {
+                "employee_id": self.employee_consultant.id,
+                "role_id": self.role_pm.id,
+                "date_start": "2022-01-01",
+                "rate": 25,
+                "sequence": 1,
+            }
+        )
+        consultant_role = self.env["hr.employee.forecast.role"].search(
+            [
+                ("employee_id", "=", self.employee_consultant.id),
+                ("role_id", "=", self.role_consultant.id),
+            ]
+        )
+        consultant_role.rate = 75
+        project_1 = self.env["project.project"].create({"name": "TestProject1"})
+        # set project in stage "to do" to get forecast
+        project_1.stage_id = self.env.ref("project.project_project_stage_0")
+        task_values = {
+            "project_id": project_1.id,
+            "forecast_role_id": self.role_consultant.id,
+            "forecast_date_planned_start": date.today(),
+            "forecast_date_planned_end": date.today(),
+            "planned_hours": 8,
+        }
+        task_values.update({"name": "Task1"})
+        task_1 = self.env["project.task"].create(task_values)
+        task_1.user_ids = self.user_consultant
+        task_values.update({"name": "Task2"})
+        task_2 = self.env["project.task"].create(task_values)
+        task_2.user_ids = self.user_consultant
+        project_2 = self.env["project.project"].create({"name": "TestProject2"})
+        # set project in stage "in progress" to get forecast
+        project_2.stage_id = self.env.ref("project.project_project_stage_1")
+        task_values.update({"project_id": project_2.id, "name": "Task3"})
+        task_3 = self.env["project.task"].create(task_values)
+        task_3.user_ids = self.user_consultant
+        task_values.update({"name": "Task4"})
+        task_4 = self.env["project.task"].create(task_values)
+        task_4.user_ids = self.user_consultant
+        forecast = self.env["forecast.line"].search(
+            [("task_id", "in", (task_1.id, task_2.id, task_3.id, task_4.id))]
+        )
+        self.assertEqual(len(forecast), 4)
+        # as we have multiple tasks to check we will divide by 4
+        self.assertAlmostEqual(sum(f.forecast_hours for f in forecast), -32.0)
+        self.assertAlmostEqual(sum(f.consolidated_forecast for f in forecast), 4.0)
+        self.assertAlmostEqual(
+            sum(f.confirmed_consolidated_forecast for f in forecast), 4.0
+        )
+        consol_employee_forecast = sum(
+            f.employee_resource_forecast_line_id.consolidated_forecast for f in forecast
+        )
+        confir_employee_forecast = sum(
+            f.employee_resource_forecast_line_id.confirmed_consolidated_forecast
+            for f in forecast
+        )
+        self.assertAlmostEqual(consol_employee_forecast, -13.0)
```

Comment:
running the test on rebase after the fix shows -16.0 here. But I don't understand what you are checking: on the sum on line 557 and 563, you are summing the same value since all the forecast lines have the same employee_resource_forecast_line_id. 

This test deserves some explanation (and likely a rewrite or a fix)

Reply:
for the record, I'm taking the opportunity of the OCA code sprint to work on this.